pkg/nodestats/nodestats.go

```diff
@@ -0,0 +1,76 @@
+package nodestats
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/iotexproject/iotex-core/blocksync"
+	"github.com/iotexproject/iotex-core/p2p"
+	"github.com/iotexproject/iotex-core/pkg/routine"
```

Comment:
the `pkg` package is typically used as a place to store reuseable and shared code, which should not directly reference other root packages (e.g. `blocksync`, `p2p`) to avoid circular dependencies. Instead, using interface could address it

Reply:
right, `pkg` should not import other like `blocksync, p2p`